pkg/services/invoice/syncada_sftp_sender.go

```
@@ -3,102 +3,65 @@ package invoice
 func InitNewSyncadaSFTPSession() (services.SyncadaSFTPSender, error) {
-	port := os.Getenv("GEX_SFTP_PORT")
-	if port == "" {
-		return nil, fmt.Errorf("Invalid credentials sftp missing GEX_SFTP_PORT")
-	}
-
-	userID := os.Getenv("GEX_SFTP_USER_ID")
+	v := viper.New()
+	v.AutomaticEnv()
+	userID := v.GetString("GEX_SFTP_USER_ID")
 	if userID == "" {
 		return nil, fmt.Errorf("Invalid credentials sftp missing GEX_SFTP_USER_ID")
 	}
 
-	remote := os.Getenv("GEX_SFTP_IP_ADDRESS")
-	if remote == "" {
-		return nil, fmt.Errorf("Invalid credentials sftp missing GEX_SFTP_IP_ADDRESS")
-	}
-
-	password := os.Getenv("GEX_SFTP_PASSWORD")
-	if password == "" {
-		return nil, fmt.Errorf("Invalid credentials sftp missing GEX_SFTP_PASSWORD")
-	}
-
-	hostKeyString := os.Getenv("GEX_SFTP_HOST_KEY")
-	if hostKeyString == "" {
-		return nil, fmt.Errorf("Invalid credentials sftp missing GEX_SFTP_HOST_KEY")
-	}
-	hostKey, _, _, _, err := ssh.ParseAuthorizedKey([]byte(hostKeyString))
-	if err != nil {
-		return nil, fmt.Errorf("Failed to parse host key %w", err)
-	}
-
 	return &SyncadaSenderSFTPSession{
-		port,
 		userID,
-		remote,
-		password,
-		hostKey,
 	}, nil
 }
 
 // SendToSyncadaViaSFTP copies specified local content to Syncada's SFTP server
 func (s *SyncadaSenderSFTPSession) SendToSyncadaViaSFTP(appCtx appcontext.AppContext, localDataReader io.Reader, syncadaFileName string) (int64, error) {
-	config := &ssh.ClientConfig{
-		User: s.userID,
-		Auth: []ssh.AuthMethod{
-			ssh.Password(s.password),
-		},
-		HostKeyCallback: ssh.FixedHostKey(s.hostKey),
-	}
+	v := viper.New()
+	v.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+	v.AutomaticEnv()
 
-	// connect
-	connection, err := ssh.Dial("tcp", s.remote+":"+s.port, config)
+	sshClient, err := cli.InitGEXSSH(appCtx, v)
 	if err != nil {
-		return 0, err
+		appCtx.Logger().Error("couldn't initialize SSH client", zap.Error(err))
```

I believe you also need to return the error here and in similar situations below